spec.html

```diff
@@ -412,25 +454,32 @@ <h1>Evaluate ( ) Concrete Method</h1>
       1. Let _module_ be this Source Text Module Record.
       1. Assert: _module_.[[Status]] is `"instantiated"` or `"evaluated"`.
       1. Let _stack_ be a new empty List.
-      1. Let _result_ be InnerModuleEvaluation(_module_, _stack_, 0).
-      1. If _result_ is an abrupt completion, then
-        1. For each module _m_ in _stack_, do
-          1. Assert: _m_.[[Status]] is `"evaluating"`.
-          1. <ins>If _m_.[[Async]] is *true*, return _result_.</ins>
-          1. Set _m_.[[Status]] to `"evaluated"`.
-          1. Set _m_.[[EvaluationError]] to _result_.
-        1. Assert: _module_.[[Status]] is `"evaluated"` and _module_.[[EvaluationError]] is _result_.
-        1. Return _result_.
-      1. Assert: _module_.[[Status]] is `"evaluated"` and _module_.[[EvaluationError]] is *undefined*.
-      1. Assert: _stack_ is empty.
-      1. <ins>If _module_.[[Async]] is *true*, return _module_.[[EvaluationPromise]].</ins>
-      1. <ins>Otherwise</ins>, return *undefined*.
+      1. <ins>If _module_.[[Async]] is *false*, then
+        1. <ins>Let _capability_ be ! NewPromiseCapability(%Promise%).</ins>
+        1. Let _result_ be ? InnerModuleEvaluation(_module_, _stack_, 0).
+        1. If _result_ is an abrupt completion, then
+          1. For each module _m_ in _stack_, do
+            1. Assert: _m_.[[Status]] is `"evaluating"`.
+            1. Set _m_.[[Status]] to `"evaluated"`.
+            1. Set _m_.[[EvaluationError]] to _result_.
+          1. Assert: _module_.[[Status]] is `"evaluated"` and _module_.[[EvaluationError]] is _result_.
+          1. <ins>Perform ! Call(_capability_.[[Reject]], *undefined*, &laquo;_result_.[[Value]]&raquo;).</ins>
+        1. <ins>Otherwise,</ins>
+          1. <ins>Perform ! Call(_capability_.[[Resolve]], *undefined*, &laquo;*undefined*&raquo;).</ins>
+        1. Assert: _module_.[[Status]] is `"evaluated"` and _module_.[[EvaluationError]] is *undefined*.
+        1. Assert: _stack_ is empty.
+        1. <ins>Return _capability_.</ins>
+      1. <ins>Otherwise,</ins>
+        1. <ins>Perform ! InnerModuleEvaluation(_module_, _stack_, 0).</ins>
+        1. <ins>Assert: _module_.[[Status]] is `"evaluated"`.</ins>
+        1. <ins>Assert: _stack_ is empty.</ins>
+        1. <ins>Return _module_.[[EvaluationPromise]].</ins>
```

Comment:
This diff is entirely for #63, and could be omitted.

Reply:
I was imagining that #63 could be a very thin layer on top, not mixed into the rest of the algorithm.